MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+
+## Motivation
+
+The current bridge design poses numerous challenges, including inefficiency, cost, and user frustration:
+
+1. **Transaction Complexity**: Requires four interactions:
+   - Two by the user (initiation and finalization).
+   - One by the relayer to complete the transaction.
+   - One by a third party (tasked by the relayer) to finalize refunds in case of failure.
+   - If any of these fails, users and the protocol face losses.
+2. **High Cost**:
+   - The multi-transaction setup is expensive for both users and the protocol.
+   - Misestimations in fee calculations could result in significant losses.
+   - Example: At 16 gwei, bridging costs $20 for 10k transactions. If fees are miscalculated by 25%, this leads to a $25k loss. At 100 gwei, the loss escalates dramatically.
+3. **Unfriendly User Experience**:
+   - The current design requires users to have funds on the destination chain for finalization. This increases friction and discourages adoption.
+   - Sponsored transactions are essential to the current implementation but remain unimplemented, leaving users stranded without funds.
+4. **Security Risks**:
+   - Relayer keys are a critical point of failure. This assumption cannot be avoided.
+   - Refund logic introduces vulnerabilities where attackers could exploit the rate limit and relayer downtime.
+   - For instance, if the relayer fails to finalize on the initiator, a malicious actor could:
+     - Take over the refund keys.
+     - Exploit both directions of the bridge (e.g., ETH→MOVE and MOVE→ETH) repeatedly, draining funds.
+5. **Maintenance Burden**:
+   - The current audit is outdated and does not reflect the significant changes made since.
+   - The current design requires a complete UI/UX overhaul, adding complexity and delay.
+6. **Unnecessary Complexity**:
+   - HTLC-based bridges are largely abandoned in favor of simpler, more effective designs.
+   - Examples like the [Consensys HTLC](https://github.com/Consensys/htlc-bridge) bridge demonstrate the [pitfalls of such approaches](https://entethalliance.org/crosschain-bridges-overview-where-we-are-now/).
+7. **Infrastructure Simplification**:
+   - Infrastructure is still incomplete for the current HTLC bridge design and we are struggling to find sound solutions that balance UI/UX and security. For example, [this issue](https://github.com/movementlabsxyz/movement/issues/838) regarding UX vs number of confirmations to require on the Ethereum side, is still under debate.
+   - Because of the over-engineered design, infrastructure is prone to error and we might end up being damaged by the amount of infrastructure we have not built yet and has to be built for the relayer to fully function.
+   - We could strip down the Relayer code and achieve a final design much more quickly.
+
+The proposed two-transaction design mitigates these issues, creating a safer, faster, and user-friendly bridge while maintaining operational reliability.
```

Comment:
This proposal is the implementation of a lock/mint bridge. 
We may reference some known explainers like [ChainLink Cross-chain bridge](https://chain.link/education-hub/cross-chain-bridge)?